xdsl/ir/core.py

```diff
@@ -641,6 +641,10 @@ def get_user_of_unique_use(self) -> Operation | None:
         return None
 
 
+_name_regex = re.compile(r"([A-Za-z_$.-][\w$.-]*)")
+_name_cleanup_pattern = re.compile(r"(_\d+)+$")
```

Comment:
Can you please add some doc strings also?
```suggestion
_VALUE_NAME_PATTERN = re.compile(r"([A-Za-z_$.-][\w$.-]*)")
_VALUE_NAME_SUFFIX_PATTERN = re.compile(r"(_\d+)+$")
```